From: Ratchanan Srirattanamet <ratchanan@ubports.com>
Date: Tue, 22 Oct 2019 03:02:47 +0700
Subject: video-format: introduce GST_VIDEO_FORMATS_ALL_WITH_ENCODED

Sometimes it's useful to specify the list of all formats that also
includes ENCODED format. For example, if an element allows passthrough,
not specifying ENCODED format will limit the usefulness of that element.

This is done by modifying original _FORMATS_ALL into a helper macro
_FORMATS_ALL_WITHOUT_BRACE to allow list expansion, then re-define
_FORMATS_ALL around that. After that, I created
_FORMATS_ALL_WITH_ENCODED (and _WITHOUT_BRACE) from it.

Forwarded: https://gitlab.freedesktop.org/gstreamer/gst-plugins-base/merge_requests/467
Last-Update: 2019-10-22

--- a/gst-libs/gst/video/video-format.h
+++ b/gst-libs/gst/video/video-format.h
@@ -490,13 +490,20 @@
 # define GST_VIDEO_OE(s) G_STRINGIFY(s)"_LE"
 #endif
 
-#define GST_VIDEO_FORMATS_ALL "{ I420, YV12, YUY2, UYVY, AYUV, RGBx, "  \
+#define GST_VIDEO_FORMATS_ALL_WITHOUT_BRACE "I420, YV12, YUY2, UYVY, AYUV, RGBx, "  \
     "BGRx, xRGB, xBGR, RGBA, BGRA, ARGB, ABGR, RGB, BGR, Y41B, Y42B, "  \
     "YVYU, Y444, v210, v216, NV12, NV21, NV16, NV61, NV24, GRAY8, GRAY16_BE, " \
     "GRAY16_LE, v308, RGB16, BGR16, RGB15, BGR15, UYVP, A420, RGB8P, YUV9, YVU9, " \
     "IYU1, ARGB64, AYUV64, r210, I420_10LE, I420_10BE, I422_10LE, I422_10BE, " \
     " Y444_10LE, Y444_10BE, GBR, GBR_10LE, GBR_10BE, NV12_64Z32, A420_10LE, "\
-    " A420_10BE, A422_10LE, A422_10BE, A444_10LE, A444_10BE }"
+    " A420_10BE, A422_10LE, A422_10BE, A444_10LE, A444_10BE"
+#define GST_VIDEO_FORMATS_ALL_WITH_ENCODED_WITHOUT_BRACE \
+    GST_VIDEO_FORMATS_ALL_WITHOUT_BRACE ", ENCODED"
+
+#define GST_VIDEO_FORMATS_ALL \
+    "{ " GST_VIDEO_FORMATS_ALL_WITHOUT_BRACE " }"
+#define GST_VIDEO_FORMATS_ALL_WITH_ENCODED \
+    "{ " GST_VIDEO_FORMATS_ALL_WITH_ENCODED_WITHOUT_BRACE " }"
 
 /**
  * GST_VIDEO_CAPS_MAKE:
